pkg/providers/instance/instance.go

```diff
@@ -161,22 +161,22 @@ func (p *DefaultProvider) BeginCreate(
 		// There may be orphan NICs (created before promise started)
 		// This err block is hit only for sync failures. Async (VM provisioning) failures will be returned by the vmPromise.Wait() function
 		if cleanupErr := p.cleanupAzureResources(ctx, GenerateResourceName(nodeClaim.Name)); cleanupErr != nil {
-			log.FromContext(ctx).Error(cleanupErr, fmt.Sprintf("failed to cleanup resources for node claim %s", nodeClaim.Name))
+			log.FromContext(ctx).Error(cleanupErr, "failed to cleanup resources for node claim", "nodeClaimName", nodeClaim.Name)
```

Comment:
Ok, hopefully the last one: For node claims (here and elsewhere), let's use "nodeClaim" key rather than "nodeClaimName". Rationale: *Name is justified for Azure resources (like VMs), which often have name and ID, but not for (cluster-scoped) Kubernetes resources.